### PR TITLE
Add support for ACK_TIMER & queued messages to the proxy

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

**Comment by peternewman** [comment]
LGTM

File: firmware/src/proxy_model.h [hunk 1]
```diff
@@ -23,8 +23,19 @@
  * @file proxy_model.h
  * @brief An RDM Model for a proxy.
  *
- * This model simulates a proxy with two responders behind it.
+ * This model simulates a proxy with two responders (children) behind it, this
+ * is a fairly typical setup when wireless DMX equipment is used.
  *
+ * The proxy will ACK_TIMER any requests sent to the child devices. The
+ * responses can then be fetched by sendint a GET QUEUED_MESSAGE to the
```
Comment (peternewman) [inline firmware/src/proxy_model.h]: SPaG sending

File: firmware/src/proxy_model.h [hunk 2]
```diff
@@ -23,8 +23,19 @@
...
+ * appropriate child device.
+ *
+ * Only a single response per-device will the buffered, if other requests are
```
Comment (peternewman) [inline firmware/src/proxy_model.h]: SPaG s/the/be/

**Comment by peternewman** [comment]
Generally LGTM. How do the RDM tests and model collector cope with the ack timer?

**Comment by nomis52** [comment]
Both the model collector and the tests work correctly.

File: firmware/src/proxy_model.h (hunk 1 above)
Comment (nomis52) [inline firmware/src/proxy_model.h]: Fixed

File: firmware/src/proxy_model.h (hunk 2 above)
Comment (nomis52) [inline firmware/src/proxy_model.h]: fixed

**Comment by peternewman** [comment]
Brilliant.

**Comment by peternewman** [comment]
There seem to still be some spelling errors.

**Comment by peternewman** [comment]
Otherwise LGTM

**Pull request merged**